Adicionando 1277 (#204)
petbccufscar/URI: Strings/1277/README.md

# Problema:

Os estudantes da tua universidade recentemente adquiriram o desagradável hábito de cabular as aulas. Para enfrentar este problema o seu Conselho de Professores decidiu somente permitir que estudantes com ao menos 75% de presença prestem os exames. A partir de uma lista de nomes de estudantes e seus respectivos registros de frequência, imprima o nome dos estudantes que não obtiveram o mínimo de presença nas aulas e que consequentemente não poderão prestar os exames.

##### Link do problema: https://www.urionlinejudge.com.br/judge/pt/problems/view/1277
 
# Resolução:

O exercício consiste em ler um número específico de casos de teste e, para cada caso de teste, imprimir quais alunos não cumpriram o mínimo de presença.

> Primeiro instanciamos as variáveis necessárias, sendo elas: três vetores do tipo `char` (onde armazenaremos o `nome`, a `frequencia` e a `saida` do programa), 7 do tipo `int` (sendo `N` o número de estudantes; `T` o número de casos de teste; `p` para somar as presenças, `a` para somar as ausências; e `i`, `j` e `k` para serem índices dos vetores) e por fim um `char` (`c` para pegar o lixo que vira do`scanf`). Em seguida lemos o `T`.
> 

Reviewer: ```suggestion
Primeiro, instanciamos as variáveis necessárias, sendo elas: três vetores do tipo `char` (onde armazenaremos o `nome`, a `frequencia` e a `saida` do programa), 7 do tipo `int` (sendo `N` o número de estudantes; `T` o número de casos de teste; `p` para somar as presenças, `a` para somar as ausências; e `i`, `j` e `k` para serem índices dos vetores) e por fim um `char` (`c` para pegar o lixo que virá do`scanf`). Em seguida, lemos o `T`.

```

```c
    int N, T;
    char nomes[5000], freq[5000], saida[5000], c;
    int i, j, k, p, a;

    scanf("%d", &T);
    scanf("%c", &c);
```

> Utilizaremos um `while(T > 0)` para passar por todos os casos de teste, sendo que sempre ao final do mesmo o T será decrementado. Dentro inicialmente iremos ler o `N` com `scanf` _e_ `nomes` e `freq` com `gets`, em sequência _'limpamos'_ a saída (marcando o índice 0 como o final utilizando o `\0`) e zeramos todas as variáveis de índice e soma.

Reviewer: ```suggestion
Utilizaremos um `while(T > 0)` para passar por todos os casos de teste, sendo que, sempre ao final do mesmo, `T` será decrementada. Dentro do loop, inicialmente, iremos ler o `N` com `scanf` e `nomes` e `freq` com `gets`, em sequência _'limpamos'_ a saída (marcando o índice 0 como o final utilizando o `\0`) e zeramos todas as variáveis de índice e soma.
```

Após cada caso individual, marcamos o final da `saida` com o `\0` e imprimimos a respectiva saída.

```c
    while (T > 0)
    {
        scanf("%d", &N);
        scanf("%c", &c);
        gets(nomes);
        gets(freq);
        saida[0] = '\0';
        i = 0;
        j = 0;
        k = 0;
        p = 0;
        a = 0;

        \\código para cada caso individual

        saida[k - 1] = '\0';
        printf("%s\n", saida);
        T--;
    }
```

> O código do caso individual será um `do{} while()` que irá verificar todas as frequências até chegar no final do respectivo vetor (ou seja, `\0`). Dentro das chaves iremos contar as presenças e faltas utilizando `if` e `else` e feito isso para um aluno iremos verificar a sua frequência e decidir se seu nome irá estar na `saida` ou não. Ao final do `do` iremos incrementar `i` que será nosso índice de `freq`.

Reviewer: ```suggestion
O código do caso individual será um `do{} while()` que irá verificar todas as frequências até chegar no final do respectivo vetor (ou seja, `\0`). Dentro das chaves iremos contar as presenças e faltas utilizando `if` e `else if`, feito isso para um aluno, iremos verificar a sua frequência e decidir se seu nome irá estar na `saida` ou não. Ao final do `do` iremos incrementar `i` que será nosso índice de `freq`.
```
